src/hooks.ts

```diff
@@ -265,24 +268,35 @@ export const useFeature: UseFeature = (featureKey, options = {}, overrides = {})
   const [prevDecisionInputs, setPrevDecisionInputs] = useState<DecisionInputs>(currentDecisionInputs);
   if (!areDecisionInputsEqual(prevDecisionInputs, currentDecisionInputs)) {
     setPrevDecisionInputs(currentDecisionInputs);
-    setDecisionState(getCurrentDecision());
+    setState(prevState => ({
+      ...prevState,
+      ...getCurrentDecision(),
+    }));
   }
 
+  const finalReadyTimeout = options.timeout !== undefined ? options.timeout : timeout;
+  useEffect(() => {
+    if (!isClientReady) {
+      subscribeToInitialization(optimizely, finalReadyTimeout, initState => {
+        setState({
+          ...getCurrentDecision(),
+          ...initState,
+        });
+      });
+    }
+  }, [isClientReady, finalReadyTimeout, setState, getCurrentDecision, optimizely]);
```

Comment:
Are we adding `setState` here because it was throwing a exhaustive dependency lint warning otherwise? If that function reference changes wondering if we could be over-eagerly running this effect.

Reply:
Good point. The lint rule seems to allow excluding it. I will remove it.